src/sgame/sg_bot_util.cpp

```diff
@@ -668,6 +679,27 @@ int BotGetDesiredBuy( gentity_t *self, weapon_t &weapon, upgrade_t upgrades[], s
 			numUpgrades ++;
 		}
 	}
+	};
+
+	if ( self->botMind->botSkillSet[BOT_H_PREFER_ARMOR] )
```

Comment:
What's the difference between BOT_H_PREFER_ARMOR vs BOT_H_BUY_ARMOR? For buy armor, I would expect a randomness factor into deciding armor if BOT_H_BUY_ARMOR is set but not B_H_PREFER_ARMOR?

Reply:
`PREFER_ARMOR` buys armor first, then maybe weapons if it can; while and `BUY_ARMOR` will buy weapon, then maybe armor if it can.

I've found that buying weapons first is a typical beginner "mistake". That's also something bots did before 0.52 or 0.53. Basically having an armor is IMO more important in combat that having a better gun.

Not knowing armor exists at all is of course an even bigger mistake, that's one more opportunity to make low skill even easier!

Reply:
hmm, in case that wasn't clear in my answer, the intent is that there are three levels in buying armor:

* first is total noob, you don't know it exists.
* second is beginner that thinks big guns are great and spends all its money on a pulse blast without buying armor. But hey if they have enough funds they'll buy an armor
* third is what they currently do, which is buying armor first. That's the best thing to do generally speaking.

Reply:
This makes sense now, but this is definitely something that should be commented in the code.

Reply:
Apparently I did put a comment on it in `src/sgame/sg_bot_local.h`, but I see how it's cut away from the code. I don't know what to do about that, the skill, the skill implementation/effect and the explanation are arguably in 3 places :-/
```diff
	BOT_A_TYRANT_CHARGE_ON_FLEE,
	BOT_H_RUN_ON_FLEE, // when fleeing, RUN
+	BOT_H_BUY_ARMOR, // knows armor exists at all
+	BOT_H_PREFER_ARMOR, // prefer to buy armor rather than guns
	BOT_H_MEDKIT, // knows the medkit even exist
```

Reply:
This is part of the reason why humans get more skill points by default. A similar very important skill is the BOT_A_AIM_HEAD one, because there is quite a radical difference in terms of inflicted damage.

Reply:
Also to avoid that, at high level, a alien bot of skill level 8 or a human bot of level 7 will always be maxed out.

Reply:
> Also to avoid that, at high level, a alien bot of skill level 8 or a human bot of level 7 will always be maxed out.

But that's dependent on some magic numbers and will change as soon as the skill list is modified.

Reply:
well, yes, the max number of skill points have to be updated too when you update the skills themselves.

Reply:
Some stats for the default human bots at skill 5, for a sample size of 63.

armor: 15 prefered armor, 35 bought armor and 13 didn't buy armor at all
medkit: 43 used medkit, while 20 pretended it doesn't exist
aim prediction: 29 did aim prediction, for 34 that didn't
running on flee: 38 did run on flee, for 25 that didn't